### DISCUSSION TITLE
How to throw error if data is invalid

### DISCUSSION THREAD

**Discussion by dominictobias-bullish** [TanStack/query#7469] [Q&A] [answered]
```ts
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      onSuccess: (data: any) => {
        // Eden Treaty doesn't throw errors I raised an issue:
        // https://github.com/elysiajs/elysia/issues/651
        // throw it ourselves so we can properly depend on `error` being filled
        if (typeof data === 'object' && data.error) {
          throw new Error(data.error)
        }
      },
      useErrorBoundary: true,
    },
  },
})
```

This doesn't work as expected because the error is actually thrown into the app. I don't want to wrap every component that uses useQuery in a boundary (and that is too broad anyway).

What I expected was that the `error` object becomes filled with the error instead of `userQuery` thinking the request was successful.

**Comment by TkDodo** [comment, marked as answer]
you would do that from within the queryFn. `onSuccess` is deprecated in v4 and was removed in v5
